### PR TITLE
Add Average Callback

### LINKED ISSUE #20617: Add Average Callback
### Description & Motivation

Create a callback to allow creating an average (smoothing) of a logged metrics and log it. 
Why directly in lightning ? This feature could be useful for Early Stopping, Now in some training the loss (for example) is very unstable and you could get an exceptional minimum, but because it's very unstable, it's not the best epochs to stop.

Get an log average feature could smooth the loss (for example again) and allow better EarlyStopping, so better training

### Pitch

Create Callback class that get in parameter : the metrics to smooth, the name of the smoothed metric and the callable function to use to smooth the metric, the window, (how many previous values to consider). 
The callable function could be mean by default and window set to 5 by default. 

### Alternatives

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by Alexis-IMBERT**
## What does this PR do?

Fixes #20617 
Create Callback class that get in parameter : the metrics to smooth, the name of the smoothed metric and the callable function to use to smooth the metric, the window, (how many previous values to consider).
The callable function could be mean by default and window set to 5 by default.

<details>
  <summary><b>Before submitting</b></summary>

- Was this **discussed/agreed** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/Lightning-AI/lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [ ] Did you make sure to **update the documentation** with your changes? (if necessary)
- [ ] Did you write any **new necessary tests**? (not for typos and docs)
- [x] Did you verify new and **existing tests pass** locally with your changes?
- [ ] Did you list all the **breaking changes** introduced by this pull request? (no breaking changes)
- [ ] Did you **update the CHANGELOG**? (not for typos, docs, test updates, or minor internal changes/refactors)

</details>

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing, make sure you have read the [review guidelines](https://github.com/Lightning-AI/lightning/wiki/Review-guidelines). In short, see the following bullet-list:

<details>
  <summary>Reviewer checklist</summary>

- [ ] Is this pull request ready for review? (if not, please submit in draft mode)
- [ ] Check that all items from **Before submitting** are resolved
- [ ] Make sure the title is self-explanatory and the description concisely explains the PR
- [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified

</details>

----
📚 Documentation preview 📚: https://pytorch-lightning--20618.org.readthedocs.build/en/20618/

### CODE REVIEW

**Comment by Alexis-IMBERT** [comment]
Please, don't hesitate to give information on things to add or change
